src/extension/message/api.ts

```diff
@@ -1,4 +1,7 @@
+import { ViewedValue } from "../viewed-state";
+
 export interface MessageToExtensionApi {
   pong: () => Promise<void>;
   openFile: (payload: { path: string; line?: number }) => Promise<void>;
+  toggleFileViewed: (payload: { path: string; value: ViewedValue }) => void;
```

Comment:
Thanks for the thorough review!

On this point, I'm thinking ahead: I wanted a boolean here for now to see it works, then change it to string so it contains a hash of the content that has been viewed, and if that content changes in the file, the extension will be able to show "changed since last viewed".

We could go with the boolean-specific `isViewed` for now and when it's hashes (if you like the use of hashes for that), we could have `viewedSha1`, what do you think?

I'm suggesting SHA1 because it's the fastest to compute, takes less space, and in this use case, doesn't have to be strongly secure against collision attacks.

Reply:
Looks good to me (`isViewed` now and later change to `viewedSha1`).